include/structures/aligned_segment.hpp

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include <seqan3/alphabet/cigar/cigar.hpp>
+
+#include "structures/breakend.hpp"          // for strand
+
+/*! \brief Read segment aligned to the reference genome (part of chimeric/split-aligned read). Contains information
+ *        parsed from the SA tag of an alignment in the SAM/BAM file.
+ *
+ * \param orientation       mapping orientation (reverse or forward strand)
+ * \param ref_name          reference/chromosome name
+ * \param pos               start position of the alignment
+ * \param mapq              mapping quality
+ * \param cig               cigar string of the alignment
+ */
+struct AlignedSegment
+{
+    strand orientation;
+    std::string ref_name;
+    int32_t pos;
+    int32_t mapq;
+    std::vector<seqan3::cigar> cig;
+
+    int32_t get_reference_start() const;
+
+    int32_t get_reference_end() const;
+
+    int32_t get_left_soft_clip() const;
+
+    int32_t get_right_soft_clip() const;
+
+    int32_t get_query_start() const;
+
+    int32_t get_query_length() const;
+
+    int32_t get_query_end() const;
```

Comment:
All member variables and function should have documentation here.

Reply:
Good point, I will open a new issue for this, as this would blow up this PR. -> https://github.com/seqan/iGenVar/issues/70